### ISSUE TITLE
Double quote character in C++ raw string literal causes Doxygen to process the string contents

### ISSUE THREAD

**Issue by jspam** [doxygen/doxygen#7265]
Take the following test code `test.cpp`

```cpp
#include <string>
#include <iostream>

int main() {
  std::string testCode(
    R"(
      // For some reason, there is a single quote here: "
      #include <bar>
      ...
    )");

  std::cout << testCode;
}
```

Note that this is valid C++ (it compiles and runs).
Execute `doxygen.exe -g Doxyfile`
Execute `doxygen.exe Doxyfile`

Expected: no warnings
Actual: warning `C:/.../doxytest/test.cpp:8: warning: include file bar not found, perhaps you forgot to add its directory to INCLUDE_PATH?`

The warning only occurs when there is a single `"` on a line. If there is a matching pair of `"` on the same line, no warning is issued.

Doxygen version: `1.8.16 (cfd73d5c4d1a66c620a3b7c08b72a3f3c3f94255*)`, OS: Windows 10

**Comment by doxygen** [comment]
I've just committed a fix for this. Please verify if it solves your problem.

**Comment by jspam** [comment]
With commit f8ce2e4edbf986248b7382a702d982173c2f41d0, no warning occurs any more, both on the test file and on the real-world source code where I encountered the issue. Thank you!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
